cpp/tests/groupby/without_agg/groupby_wo_agg_test.cu

```diff
@@ -0,0 +1,793 @@
+  // // The aggregation type is passed via a member of the template argument class
+  // const agg_op op = test_parameters::op;
+
+  const GroupByOutType group_output_type {test_parameters::group_output_type};
+
+  gdf_context ctxt;
+ 
+  // multi_column_t is a tuple of vectors. The number of vectors in the tuple
+  // determines the number of columns to be grouped, and the value_type of each
+  // vector determiens the data type of the column
+  using multi_column_t = typename test_parameters::multi_column_t;
+
+  //output_t is the output type of the aggregation column
+  using output_t = gdf_size_type;
+
+  //map_t is used for reference solution
+  using map_t = typename test_parameters::ref_map_type;
+
+  //tuple_t is tuple of datatypes associated with each column to be grouped
+  using tuple_t = typename test_parameters::tuple_t;
+
+  //contains input generated for gdf calculation and reference solution
+  multi_column_t input_key;
+
+  //contains the input aggregation column
+  std::vector<output_t> input_value;
+
+  //contains grouped by column output of the gdf groupby call
+  multi_column_t cpu_data_cols_out;
+
+  //contains the aggregated output column
+  std::vector<gdf_size_type> cpu_out_indices;
+
+  // Type for a unique_ptr to a gdf_column with a custom deleter
+  // Custom deleter is defined at construction
+  using gdf_col_pointer = typename std::unique_ptr<gdf_column, std::function<void(gdf_column*)>>;
+
+  // Containers for unique_ptrs to gdf_columns that will be used in the gdf_group_by functions
+  // unique_ptrs are used to automate freeing device memory
+  std::vector<gdf_col_pointer> gdf_input_key_columns;
+  gdf_col_pointer gdf_input_value_column;
+
+  std::vector<gdf_col_pointer> gdf_cpu_data_cols_out_columns;
+  // gdf_col_pointer gdf_output_indices_column;
+  rmm::device_vector<gdf_size_type> gdf_output_indices_column;
+
+  // Containers for the raw pointers to the gdf_columns that will be used as input
+  // to the gdf_group_by functions
+  std::vector<gdf_column*> gdf_raw_input_key_columns;
+  gdf_column* gdf_raw_input_val_column;
+  std::vector<gdf_column*> gdf_raw_gdf_data_cols_out_columns;
+  // gdf_column* gdf_raw_out_indices;
+  gdf_size_type* gdf_raw_out_indices;
+  gdf_size_type gdf_raw_out_indices_size;
+
+  GroupByWoAggTest()
+  {
+    // Use constant seed so the psuedo-random order is the same each time
+    // Each time the class is constructed a new constant seed is used
+    static size_t number_of_instantiations{0};
+    std::srand(number_of_instantiations++);
+
+    if (this->group_output_type == GroupByOutType::SQL) {
+      ctxt.flag_groupby_include_nulls = true;
+      ctxt.flag_nulls_sort_behavior = GDF_NULL_AS_LARGEST;
+    } else {
+      ctxt.flag_groupby_include_nulls = false;
+      ctxt.flag_nulls_sort_behavior = GDF_NULL_AS_LARGEST;
+    }
+  }
+
+  ~GroupByWoAggTest()
+  {
+  }
+
+  template <typename col_type>
+  gdf_col_pointer create_gdf_column(std::vector<col_type> const & host_vector,
```

Comment:
Could you simplify by using the new column_wrapper class from PR #694 ?

Reply:
No response?

Reply:
Deleted it and used existing test utilities